source/event_stream_rpc.c

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/event-stream/event_stream_rpc.h>
+
+#include <inttypes.h>
+
+const struct aws_byte_cursor aws_event_stream_rpc_message_type_name =
+    AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(":message-type");
+const struct aws_byte_cursor aws_event_stream_rpc_message_flags_name =
+    AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(":message-flags");
+const struct aws_byte_cursor aws_event_stream_rpc_stream_id_name = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(":stream-id");
+const struct aws_byte_cursor aws_event_stream_rpc_operation_name = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("operation");
+
+/* just a convenience function for fetching message metadata from the event stream headers on a single iteration. */
+int aws_event_stream_rpc_fetch_message_metadata(
```

Comment:
nitpick: more like `extract`/`read`/`parse`/`decode`, fetch kind of implies it has to go somewhere to get it